types/ember__owner/index.d.ts

```diff
@@ -0,0 +1,110 @@
+// Type definitions for non-npm package @ember/owner 4.0
+// Project: https://emberjs.com/api/ember/4.0/
+// Definitions by: Chris Krycho <https://github.com/chriskrycho>
+//                 Dan Freeman <https://github.com/dfreeman>
+//                 James C. Davis <https://github.com/jamescdavis>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.4
+
+/**
+ * The name for a factory consists of a namespace and the name of a specific
+ * type within that namespace, like `'service:session'`.
+ */
+type FullName = `${string}:${string}`;
+
+// TODO: when migrating into Ember proper, evaluate whether we should introduce
+// a registry which users can provide to resolve known types, so e.g.
+// `owner.lookup('service:session')` can return the right thing.
```

Comment:
The nice thing about finally having a dedicated `Owner` type is that folks can play around with this themselves locally via declaration merging if they like 👍

Reply:
Indeed! If/as we publish a blog post about this, we should actually explicitly show people how to do that with an eye to getting some feedback about where it falls over.